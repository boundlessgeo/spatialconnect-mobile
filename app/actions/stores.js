'use strict';
import { stores } from 'spatialconnect/native';

export const loadStores = () => {
  return dispatch => {
    stores().take(1).subscribe(data => {
      dispatch({
        type: 'ADD_STORE_LIST',
        stores: data.stores
      });
    });
  };
};