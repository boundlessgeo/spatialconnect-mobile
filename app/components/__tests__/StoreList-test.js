/*global jest,describe,it,expect*/
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import StoreList from '../StoreList';
import stores from 'stores';

describe('<StoreList />', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <StoreList stores={stores} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});