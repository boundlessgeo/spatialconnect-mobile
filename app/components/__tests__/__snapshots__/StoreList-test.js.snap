exports[`<StoreList /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e4e7ec",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 0
    }
  }>
  <ListView
    dataSource={
      ListViewDataSource {
        "_cachedRowCount": 5,
        "_dataBlob": Object {
          "s1": Array [
            Object {
              "default_layers": Array [],
              "id": "3dc5afc9-393b-444c-8581-582e2c2d98a3",
              "name": "gj1",
              "store_type": "geojson",
              "uri": "simple.geojson",
              "version": "1"
            },
            Object {
              "default_layers": Array [],
              "id": "3e6c072e-8e62-41be-8d1a-7a3116df9c16",
              "name": "gj2",
              "store_type": "geojson",
              "uri": "feature.geojson",
              "version": "1"
            },
            Object {
              "default_layers": Array [],
              "id": "f6dcc750-1349-46b9-a324-0223764d46d1",
              "name": "Haiti",
              "store_type": "gpkg",
              "uri": "http://www.geopackage.org/data/haiti-vectors-split.gpkg",
              "version": "1"
            },
            Object {
              "default_layers": Array [],
              "id": "fad33ae1-f529-4c79-affc-befc37c104ae",
              "name": "Whitehorse",
              "store_type": "gpkg",
              "uri": "https://portal.opengeospatial.org/files/63156",
              "version": "1"
            },
            Object {
              "default_layers": Array [
                "spatialconnect:baseball_team"
              ],
              "id": "71522e9b-3ec6-48c3-8d5c-57c8d14baf6a",
              "name": "wfstest",
              "store_type": "wfs",
              "uri": "http://efc-dev.boundlessgeo.com:8080/geoserver/spatialconnect/ows",
              "version": "1.1.0"
            }
          ]
        },
        "_dirtyRows": Array [
          Array [
            true,
            true,
            true,
            true,
            true
          ]
        ],
        "_dirtySections": Array [
          true
        ],
        "_getRowData": [Function defaultGetRowData],
        "_getSectionHeaderData": [Function defaultGetSectionHeaderData],
        "_rowHasChanged": [Function rowHasChanged],
        "_sectionHeaderHasChanged": [Function anonymous],
        "rowIdentities": Array [
          Array [
            "0",
            "1",
            "2",
            "3",
            "4"
          ]
        ],
        "sectionIdentities": Array [
          "s1"
        ]
      }
    }
    enableEmptySections={true}
    renderRow={[Function bound renderRow]}
    renderSeparator={[Function bound renderSeparator]}
    style={Object {}} />
</View>
`;
